redisinsight/api/src/modules/settings/settings.service.ts

```diff
@@ -56,16 +56,40 @@ export class SettingsService {
   ): Promise<GetAppSettingsResponse> {
     this.logger.debug('Getting application settings.', sessionMetadata);
     try {
-      const agreements =
-        await this.agreementRepository.getOrCreate(sessionMetadata);
       const settings =
         await this.settingsRepository.getOrCreate(sessionMetadata);
+
+      let agreements;
+      if (SERVER_CONFIG.acceptTermsAndConditions) {
+        const isEncryptionAvailable =
+          (await this.keyEncryptionStrategy.isAvailable()) ||
+          (await this.keytarEncryptionStrategy.isAvailable());
```

Comment:
let's use this method https://github.com/RedisInsight/RedisInsight/blob/main/redisinsight/api/src/modules/encryption/encryption.service.ts#L28 and check it's output or even better to add additional function to that service e.g. `isEncryptionAvailable()` which will return boolean flag based on own method. so all logic will be in a single place

Reply:
Looks good